docs/guides/developers.md

```diff
@@ -0,0 +1,45 @@
+# Developer guide
```

Comment:
maybe we should discuss how we want to extend the developers guide and make specific tasks for that?

Reply:
currently I am missing from the issue description:

How to test new models.
-- When do we use files for test data and when do we use in-memory data.

Reply:
I will indeed update the issue. Good point! I know I have said it before, but I think we should start creating more specific issues when we pick up the documentation overhaul!